src/database/remove.ts

```diff
@@ -1,99 +1,108 @@
    * Construct a new RTDB delete operation.
    *
    * @constructor
+   * @param instance RTBD instance ID.
    * @param path path to delete.
-   * @param options
    */
-  constructor(path: string, options: DatabaseRemoveOptions) {
+  constructor(instance: string, path: string) {
     this.path = path;
-    this.concurrency = options.concurrency;
-    this.retries = options.retries;
-    this.remote = new RTDBRemoveRemote(options.instance);
-    this.waitingPath = new Map();
-    this.jobStack = new Stack<string, void>({
-      name: "long delete stack",
-      concurrency: this.concurrency,
-      handler: this.chunkedDelete.bind(this),
-      retries: this.retries,
+    this.remote = new RTDBRemoveRemote(instance);
+    this.deleteJobStack = new Stack({
+      name: "delete stack",
+      concurrency: 1,
+      retries: 3,
+    });
+    this.listStack = new Stack({
+      name: "list stack",
+      concurrency: 1,
+      retries: 3,
     });
   }
 
-  execute(): Promise<void> {
-    const prom: Promise<void> = this.jobStack.wait();
-    this.jobStack.add(this.path);
-    return prom;
+  async execute(): Promise<void> {
+    await this.deletePath(this.path);
   }
 
-  private chunkedDelete(path: string): Promise<void> {
-    return this.remote
-      .prefetchTest(path)
-      .then((test: NodeSize) => {
-        switch (test) {
-          case NodeSize.SMALL:
-            return this.remote.deletePath(path);
-          case NodeSize.LARGE:
-            return this.remote.listPath(path).then((pathList: string[]) => {
-              if (pathList) {
-                for (const p of pathList) {
-                  this.jobStack.add(pathLib.join(path, p));
-                }
-                this.waitingPath.set(path, pathList.length);
-              }
-              return false;
-            });
-          case NodeSize.EMPTY:
-            return true;
-          default:
-            throw new FirebaseError("Unexpected prefetch test result: " + test, { exit: 3 });
-        }
-      })
-      .then((deleted: boolean) => {
-        if (!deleted) {
-          return;
-        }
-        if (path === this.path) {
-          this.jobStack.close();
-          logger.debug("[database][long delete stack][FINAL]", this.jobStack.stats());
-        } else {
-          const parentPath = pathLib.dirname(path);
-          const prevParentPathReference = this.waitingPath.get(parentPath);
-          if (!prevParentPathReference) {
-            throw new FirebaseError(
-              `Unexpected error: parent path reference is zero for path=${path}`,
-              { exit: 3 }
-            );
-          }
-          this.waitingPath.set(parentPath, prevParentPathReference - 1);
-          if (this.waitingPath.get(parentPath) === 0) {
-            this.jobStack.add(parentPath);
-            this.waitingPath.delete(parentPath);
-          }
+  /**
+   * @return true if this path is small (i.e. can be deleted with a single request with writeSizeLimit=tiny),
```

Comment:
This doesn't _just_ return true if it's small, right? Can you add more detail to what this function does in the documentation? I think this might want to be broken up, but it's not clear to me what the goal is just looking at this function...

Reply:
Sure.
I explained the algorithm in the jsdoc.